store/sqlstore/tokens_store.go

```diff
@@ -32,33 +33,33 @@ func newSqlTokenStore(sqlStore SqlStore) store.TokenStore {
 func (s SqlTokenStore) createIndexesIfNotExists() {
 }
 
-func (s SqlTokenStore) Save(token *model.Token) *model.AppError {
+func (s SqlTokenStore) Save(token *model.Token) error {
 	if err := token.IsValid(); err != nil {
 		return err
 	}
 
 	if err := s.GetMaster().Insert(token); err != nil {
-		return model.NewAppError("SqlTokenStore.Save", "store.sql_recover.save.app_error", nil, "", http.StatusInternalServerError)
+		return errors.Wrap(err, "failed to save Token")
 	}
 	return nil
 }
 
-func (s SqlTokenStore) Delete(token string) *model.AppError {
+func (s SqlTokenStore) Delete(token string) error {
 	if _, err := s.GetMaster().Exec("DELETE FROM Tokens WHERE Token = :Token", map[string]interface{}{"Token": token}); err != nil {
-		return model.NewAppError("SqlTokenStore.Delete", "store.sql_recover.delete.app_error", nil, "", http.StatusInternalServerError)
+		return errors.Wrapf(err, "failed to delete Token with value %s", token)
 	}
 	return nil
 }
 
-func (s SqlTokenStore) GetByToken(tokenString string) (*model.Token, *model.AppError) {
+func (s SqlTokenStore) GetByToken(tokenString string) (*model.Token, error) {
 	token := &model.Token{}
 
 	if err := s.GetReplica().SelectOne(token, "SELECT * FROM Tokens WHERE Token = :Token", map[string]interface{}{"Token": tokenString}); err != nil {
 		if err == sql.ErrNoRows {
-			return nil, model.NewAppError("SqlTokenStore.GetByToken", "store.sql_recover.get_by_code.app_error", nil, err.Error(), http.StatusBadRequest)
+			return nil, store.NewErrInvalidInput("Token", "Token", tokenString)
```

Comment:
This should be a NotFound error. Let's make that change.

Reply:
Yep, I was not sure about that because as you suggest the token was not found but it was considered as a bad input, that's why a kept that response and I wanted to preserve the current behaviour without change the http codes

Reply:
Yes, I think it should be fine. 404 is the right response to return here.